### PR TITLE
Adding end to end tests for Spark SQL and DF. 

### PR DESCRIPTION
**Pull request by ahmadnazeri**
# Description

This PR fixes some issues with supporting data frames and adds spark end to end testing.

### Does this correspond to an open issue?

n/a

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have fixed any merge conflicts

### CODE REVIEW

File: provider/spark.go [hunk 1]
```diff
@@ -1782,6 +1813,36 @@ func (s *S3Store) ResourceStreamSingleFile(key string, begin int64) (chan interf
...
+	filePath := fileLocation[prefixLength:]
+
+	sess, err := session.NewSession(
+		&awsV1.Config{
+			Region:      awsV1.String("us-east-2"),
+			Credentials: s.credentials,
+		},
+	)
+	if err != nil {
+		panic(err)
```
Comment (sdreyer) [inline provider/spark.go:1831]: Change this to return error rather than panic

File: provider/spark.go [hunk 2]
```diff
@@ -1782,6 +1813,36 @@ func (s *S3Store) ResourceStreamSingleFile(key string, begin int64) (chan interf
...
+	}
+
+	uploader := s3manager.NewUploader(sess)
+	_, err = uploader.Upload(&s3manager.UploadInput{
+		Bucket: aws.String(s.bucketPath), // Bucket to be used
+		Key:    aws.String(filePath),     // Name of the file to be saved
+		Body:   file,                     // File
+	})
+	if err != nil {
+		panic(err)
```
Comment (sdreyer) [inline provider/spark.go:1841]: error instead of panic

File: provider/spark.go [hunk 3]
```diff
@@ -1782,6 +1813,36 @@ func (s *S3Store) ResourceStreamSingleFile(key string, begin int64) (chan interf
 	return rowChannel, nil
 }
 
+func (s *S3Store) GetTransformationFileLocation(id ResourceID) string {
+	return fmt.Sprintf("s3://%s/featureform/DFTransformations/%s/%s", s.bucketPath, id.Name, id.Variant)
```
[the remark this answers is not in the document]
Comment (ahmadnazeri) [reply provider/spark.go:1817]: This is just used for Transformation File (i.e. for data frames one)

[the remark this answers is not in the document]
Comment (sdreyer) [reply provider/spark.go:1817]: gotcha

**Review by sdreyer — APPROVED** [review]

**Pull request merged**